hazelcast/src/main/java/com/hazelcast/internal/management/dto/ClientBwListEntryDTO.java

```diff
@@ -51,9 +59,52 @@ public void fromJson(JsonObject json) {
         value = getString(json, "value");
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ClientBwListEntryDTO that = (ClientBwListEntryDTO) o;
+
+        if (type != that.type) {
+            return false;
+        }
+        return value.equals(that.value);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = type.hashCode();
+        result = 31 * result + value.hashCode();
+        return result;
+    }
+
     public enum Type {
 
-        IP_ADDRESS, INSTANCE_NAME, LABEL
+        IP_ADDRESS(0), INSTANCE_NAME(1), LABEL(2);
+
+        private final int id;
+
+        Type(int id) {
+            this.id = id;
+        }
+
+        public int getId() {
+            return id;
+        }
+
+        public static Type getById(final int id) {
+            for (Type type : values()) {
+                if (type.id == id) {
+                    return type;
+                }
+            }
+            return null;
```

Comment:
nit: can we please implement these simple lookups with stream api?

Reply:
We keep the code as low-level and lightweight as it possible in the IMDG core, so this loop is the preferred implementation.